src/routes/auth/guards/auth.guard.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Inject,
+  Injectable,
+} from '@nestjs/common';
+import { IAuthRepository } from '@/domain/auth/auth.repository.interface';
+import { IJwtService } from '@/datasources/jwt/jwt.service.interface';
```

Comment:
To keep the separation of responsibilities between the layers, this should call a domain handler.

Reply:
I added a new `AuthRepository['verifyAccessToken']` method in b5aed5145112dbbc6648486f44d67c343fb43d7c which is now called here instead.